presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -1400,6 +1462,195 @@ public synchronized void progressChanged(ProgressEvent progressEvent)
         }
     }
 
+    private static class PrestoS3StreamingOutputStream
+            extends OutputStream
+    {
+        private final AmazonS3 s3;
+        private final String bucketName;
+        private final String key;
+        private final String uploadId;
+
+        private final ListeningExecutorService uploadExecutor;
+
+        private int currentPartNumber;
+        private byte[] buf;
+        private int count;
+
+        private ListenableFuture<UploadPartResult> inProgressUploadFuture;
+        private final List<UploadPartResult> parts;
+
+        public PrestoS3StreamingOutputStream(
+                AmazonS3 s3,
+                String bucketName,
+                String key,
+                String uploadId,
+                ListeningExecutorService uploadExecutor,
+                long multiPartUploadMinPartSize)
+        {
+            STATS.uploadStarted();
+
+            this.s3 = requireNonNull(s3, "s3 is null");
+
+            int bufferSize = (int) multiPartUploadMinPartSize;
+            this.buf = new byte[bufferSize];
+
+            this.bucketName = bucketName;
```

Comment:
Add `requireNonNull` for all of these

(this is a private class, so not strictly necessary, but best to be consistent)